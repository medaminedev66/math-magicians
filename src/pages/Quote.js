import React from 'react';

const Quote = () => (
  <h1 className="Quote">
    Mathematics is not about numbers, equations, computations, or algorithms:
    it&lsquo;s about understanding.--WIlliam Paul Thurston
  </h1>
);

export default Quote;
