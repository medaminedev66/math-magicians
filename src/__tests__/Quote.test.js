import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Quote from '../pages/Quote';

describe('test the Quote components', () => {
  it('renders without errors', () => {
    const root = document.createElement('div');
    ReactDOM.render(<Quote />, root);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { queryByTitle } = render(<Quote />);
    expect(queryByTitle('quote').innerHTML).toContain(
      'Mathematics is not about numbers',
    );
  });
});
