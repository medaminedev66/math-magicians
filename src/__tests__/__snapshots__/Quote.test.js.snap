// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Quote components matches  1`] = `
<h1
  className="Quote"
  title="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it‘s about understanding.--WIlliam Paul Thurston
</h1>
`;

exports[`test the Quote components matches the snapshot 1`] = `
<h1
  className="Quote"
  title="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it‘s about understanding.--WIlliam Paul Thurston
</h1>
`;

exports[`test the Quote components renders correctly 1`] = `
<h1
  className="Quote"
  title="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it‘s about understanding.--WIlliam Paul Thurston
</h1>
`;
